addons/web/controllers/main.py

```diff
@@ -325,18 +325,29 @@ def set_cookie_and_redirect(redirect_url):
     return redirect
 
 def clean_action(action, env):
-    action.setdefault('flags', {})
     action_type = action.setdefault('type', 'ir.actions.act_window_close')
     if action_type == 'ir.actions.act_window':
         action = fix_view_modes(action)
 
     # When returning an action, only few information are really usefull
-    return {
+    readable_fields = env[action['type']]._get_readable_fields()
+    action_type_fields = set(env[action['type']]._fields)
```

Comment:
```suggestion
    action_type_fields = env[action['type']]._fields.keys()
```
`dict_keys` is a set, so the conversion should, strictly speaking, not be necessary.